### PR TITLE
Crud endpoints plants

### PR DESCRIPTION
**Pull request by D-Lessenden**
Create CRUD endpoints for plants. For sad-path testing I followed Hash's lead for the CRUD user endpoints and return a 204 No Content.

### CODE REVIEW

File: app/serializers/plant_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class PlantSerializer
```
Comment (moosehandlr) [inline app/serializers/plant_serializer.rb:1]: Lets include relationships here at the bottom.
`has_many :garden_plants
  has_many :gardens, through: :garden_plants`

File: app/serializers/plant_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  include FastJsonapi::ObjectSerializer
+  attributes :image
+  attributes :name
+  attributes :species
+  attributes :description
+  attributes :light_requirements
+  attributes :water_requirements
+  attributes :when_to_plant
+  attributes :harvest_time
+  attributes :common_pests
```
Comment (moosehandlr) [inline app/serializers/plant_serializer.rb:11]: Let's include plant relationships here at the bottom.
`has_many :garden_plants`
 `has_many :gardens, through: :garden_plants`

**Review by moosehandlr — CHANGES_REQUESTED** [review]
Comments in serializer for needed changes. Everything else looks good to me! Solid work.

File: app/serializers/plant_serializer.rb (hunk 2 above)
Comment (D-Lessenden) [reply app/serializers/plant_serializer.rb:11]: My local branch was behind the remote branch. I made the change by editing the file here on GH,

**Comment by moosehandlr** [comment]
Look great!

**Pull request merged**